lib/progress_bar.ex

```diff
@@ -22,4 +22,32 @@ defmodule ProgressBar do
   def render_spinner(custom_format, fun) do
     ProgressBar.Spinner.render(custom_format, fun)
   end
+
+  def from_enum(list, fun, opts \\ []) do
+    total = Enum.count(list)
+
+    ProgressBar.render(0, total, opts)
+
+    list
+    |> Enum.with_index(1)
+    |> Enum.each(fn {el, i} ->
+      fun.(el)
+      ProgressBar.render(i, total, opts)
+    end)
+
+    list
+  end
+
+  def from_steam(list, opts \\ []) do
+    total = Enum.count(list)
+
+    ProgressBar.render(0, total, opts)
+
+    list
+    |> Stream.with_index(1)
+    |> Stream.map(fn {el, i} ->
```

Comment:
Could this be simplified to `Stream.each`?

Reply:
Using `Stream.each` would not allow this function to be used in between pipes.

Reply:
Hm, how do you mean? It still returns a stream.

```
iex(7)> [10, 20, 30] |> Stream.with_index(1) |> Stream.each(fn {el, i} -> IO.puts("El #{el} with index #{i}.") end) |> Stream.map(fn {el, i} -> {el * 2, i} end) |> Enum.to_list()
El 10 with index 1.
El 20 with index 2.
El 30 with index 3.
[{20, 1}, {40, 2}, {60, 3}]
```

Reply:
Ah, the map acts to strip away the index of course. Never mind, then.

Reply:
Wait, really? I was confusing it with `Enum.each` then I guess, which returns `:ok` everytime. So nevermind my comment.

Reply:
Anyway, yeah, the map is still needed, cause we need to remove the index.